integration-tests/src/mock_network/mod.rs

```diff
@@ -0,0 +1,304 @@
+        }
+        self.network_info.highest_height_peers = self.network_info.connected_peers.clone();
+        near_performance_metrics::actix::run_later(
+            ctx,
+            self.block_production_delay,
+            move |act, ctx| {
+                act.update_peers(ctx);
+            },
+        );
+    }
+}
+
+impl Actor for MockPeerManagerActor {
+    type Context = Context<Self>;
+    fn started(&mut self, ctx: &mut Self::Context) {
+        // Start syncing job.
+        self.update_peers(ctx);
+    }
+}
+
+impl Handler<PeerManagerMessageRequest> for MockPeerManagerActor {
+    type Result = PeerManagerMessageResponse;
+
+    fn handle(&mut self, msg: PeerManagerMessageRequest, _ctx: &mut Self::Context) -> Self::Result {
+        match msg {
+            PeerManagerMessageRequest::NetworkRequests(request) => match request {
+                NetworkRequests::BlockRequest { hash, peer_id } => {
+                    let block = self.chain_history_access.retrieve_block(&hash).unwrap();
+                    let _response = self
+                        .client_addr
+                        .do_send(NetworkClientMessages::Block(block, peer_id, true));
+                }
+                NetworkRequests::BlockHeadersRequest { hashes, peer_id } => {
+                    let headers =
+                        self.chain_history_access.retrieve_block_headers(hashes.clone()).unwrap();
+                    let _response = self
+                        .client_addr
+                        .do_send(NetworkClientMessages::BlockHeaders(headers, peer_id));
+                }
+                NetworkRequests::PartialEncodedChunkRequest { request, .. } => {
+                    let response =
+                        self.chain_history_access.retrieve_partial_encoded_chunk(&request).unwrap();
+                    let _response = self
+                        .client_addr
+                        .do_send(NetworkClientMessages::PartialEncodedChunkResponse(response));
+                }
+                NetworkRequests::Block { .. } => {}
+                _ => {
+                    panic!("MockPeerManagerActor receives unexpected message {:?}", request);
+                }
+            },
+            _ => {
+                panic!("MockPeerManagerActor receives unexpected message {:?}", msg);
+            }
+        }
+        PeerManagerMessageResponse::NetworkResponses(NetworkResponses::NoResponse)
+    }
+}
+
+#[derive(actix::Message, Debug)]
+#[rtype(result = "u64")]
+pub struct GetChainHistoryFinalBlockHeight;
+
+impl Handler<GetChainHistoryFinalBlockHeight> for MockPeerManagerActor {
+    type Result = u64;
+
+    fn handle(
+        &mut self,
+        _msg: GetChainHistoryFinalBlockHeight,
+        _ctx: &mut Self::Context,
+    ) -> Self::Result {
+        self.chain_history_access.chain.head().unwrap().height
+    }
+}
+
+/// This class provides access a pre-generated chain history
+struct ChainHistoryAccess {
+    chain: Chain,
+}
```

Comment:
feels like you only need `ChainStore` here

Reply:
Actually `Chain` is needed because some functions here such as `retrieve_headers` do rely on logic that is implemented in `Chain` not `ChainStore`.